### PR TITLE
[feature/task2]: Implemented components for Fantasy Game

### PR DESCRIPTION
**Pull request by CherpakovPhilipp**

### CODE REVIEW

File: src/2-building-blocks-of-oop-part-1/Item.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { Comparable } from './Comparable';
+
+let id = 0;
```
Comment (dianababurina) [inline src/2-building-blocks-of-oop-part-1/Item.ts:3]: storing values as global vars is not a good practice

File: src/2-building-blocks-of-oop-part-1/Item.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    return this.value;
+  }
+
+  getName(): string {
+    return this.name;
+  }
+
+  getWeight(): number {
+    return this.weight;
+  }
```
Comment (dianababurina) [inline src/2-building-blocks-of-oop-part-1/Item.ts:44]: can't we use getters here?

File: src/2-building-blocks-of-oop-part-1/Item.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    this.value = value;
+  }
+
+  setName(name: string): void {
+    this.name = name;
+  }
+  
+  setWeight(weight: number): void {
+    this.weight = weight;
+  }
```
Comment (dianababurina) [inline src/2-building-blocks-of-oop-part-1/Item.ts:56]: can't we use setters here?

**Review by dianababurina — APPROVED** [review]

File: src/2-building-blocks-of-oop-part-1/Item.ts (hunk 1 above)
Comment (CherpakovPhilipp) [reply src/2-building-blocks-of-oop-part-1/Item.ts:3]: yes, I agree, but it was one of the tasks criteria

**Pull request merged**